src/utils/use-reduced-motion.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState, useContext } from "react"
+import {
+    prefersReducedMotion,
+    determineShouldReduceMotion,
+} from "../dom/accessibility"
+import { MotionContext } from "../motion/context/MotionContext"
+
+/**
+ * A hook that returns `true` if we should be using reduced motion based on the current device's Reduced Motion setting.
+ *
+ * This can be used to implement changes to your UI based on Reduced Motion. For instance, replacing motion-sickness inducing
+ * `x`/`y` animations with `opacity, disabling the autoplay of background videos, or turning off parallax motion.
+ *
+ * It will actively respond to changes and re-render your components with the latest setting.
+ *
+ * ```jsx
+ * export function Sidebar({ isOpem }) {
+ *   const shouldReduceMotion = useReducedMotion()
+ *   const closedX = shouldReduceMotion ? 0 : "-100%"
+ *
+ *   return (
+ *     <motion.div animate={{
+ *       opacity: isOpen ? 1 : 0,
+ *       x: isOpen ? 0 : closedX
+ *     }} />
+ *   )
+ * }
+ * ```
+ *
+ * @return boolean
+ */
+export function useReducedMotion() {
+    const { isReducedMotion } = useContext(MotionContext)
+    const [shouldReduceMotion, setShouldReduceMotion] = useState(
+        determineShouldReduceMotion(prefersReducedMotion.get(), isReducedMotion)
+    )
+
+    useEffect(
+        () => {
+            return prefersReducedMotion.onChange(() => {
```

Comment:
I'm not too familiar with motionValue API and I'm removing this from GitHub so I can't quite check right now type signature, but seems a little bit weird to call `prefersReducedMotion.get()` here, isnt current value provided to the onChange callback?

Reply:
yeah i was just keeping it the same as above but this is a good point